addons/web_editor/static/src/js/editor/odoo-editor/test/spec/editor.test.js

```diff
@@ -2297,6 +2353,101 @@ X[]
                     });
                 });
             });
+            describe('Nested editable zone (inside contenteditable=false element)', () => {
+                it('should remove the uneditable nesting zone if the last element of the nested zone is empty', async () => {
+                    await testEditor(BasicEditor, {
+                        contentBefore: unformat(`
+                            <div contenteditable="false">
+                                <div contenteditable="true">
+                                    <p>[]<br></p>
+                                </div>
+                            </div>
+                        `),
+                        stepFunction: deleteBackward,
+                        contentAfter: unformat(`
+                            <p>[]<br></p>
+                        `),
+                    });
+                });
+                it('should not remove the last element of a nested zone if not empty', async () => {
+                    await testEditor(BasicEditor, {
+                        contentBefore: unformat(`
+                            <div contenteditable="false">
+                                <div contenteditable="true">
+                                    <p>[]content</p>
+                                </div>
+                            </div>
+                        `),
+                        stepFunction: deleteBackward,
+                        contentAfter: unformat(`
+                            <div contenteditable="false">
+                                <div contenteditable="true">
+                                    <p>[]content</p>
+                                </div>
+                            </div>
+                        `),
+                    });
+                });
+                it('should move a line before the uneditable nesting zone if the cursor is at the start of the first element of the nested zone', async () => {
+                    await testEditor(BasicEditor, {
+                        contentBefore: unformat(`
+                            <div contenteditable="false">
+                                <div contenteditable="true">
+                                    <p>[]content</p>
+                                    <p>othercontent</p>
+                                </div>
+                            </div>
+                        `),
+                        stepFunction: deleteBackward,
+                        contentAfter: unformat(`
+                            <p>[]content</p>
+                            <div contenteditable="false">
+                                <div contenteditable="true">
+                                    <p>othercontent</p>
+                                </div>
+                            </div>
+                        `),
+                    });
+                });
```

Comment:
All those behaviors are weird from my perspective. I feel like this should be discussed with more people from the editor team too.

EDIT: They actually match the spec. It's just a little complicated to wrap one's head around the concepts in practice without the actual case in knowledge. The screenshots in the spec really help.

However, I can't help but feeling this might not match the expected behavior in the classic, non-knowledge case of non-editable contexts. I feel like when I set `contentEditable="false"` then `contentEditable="true"`, I don't want the content inside to escape outside of the boundaries that I have so carefully created.

I feel like in knowledge those boundaries only exist for technical reasons (prevent the browser from messing with knowledge nodes) while in other cases you might want to actually set editable context boundaries that should not be breached. 

I feel like we might need a new concept for nodes that are protected for technical reasons, non-editable-then-editable for technical reasons, but are otherwise to be treated like normal containers. Because that's basically what you're doing here, re-implementing delete operations on those nodes to behave almost exactly like if they were simple paragraphs. Maybe we need to introduce something like a class or so to enable this behavior, by marking those `contentEditable=false` as somewhat "permeable" ?

Reply:
I should test this, but in cases you describe in website, isn't the `odoo-editor-editable` always `contenteditable="false"` ? Subsections are then set as `contenteditable="true", and this means that what we are searching for is a node inside such a section which is `contenteditable="false"`, contains editable zones and those editable zones should never escape that `contenteditable` context

I'm not seeing such an element at first glance, but I understand your concern, maybe we could combine the feature implemented in this PR with the already existing `data-oe-protected` attribute ? 

we would only allow navigating through `contenteditable="false"` elements that are in between a `data-oe-protected="true"` node and a `data-oe-protected="false"` one ?